src/iSparrow/sparrow_watcher.py

```diff
@@ -0,0 +1,527 @@
+        pattern: str = ".wav",
+        check_time: int = 1,
+        delete_recordings: str = "on_cleanup",
+        reanalyze_on_cleanup: bool = True,
+    ):
+        """
+        __init__ Create a new Watcher object.
+
+        Args:
+            indir (str): Directory to read files from
+            outdir (str): Directory to write the analysis results to.
+                        A new directory with a timestamp for name will be
+                        created within it to store the result files in.
+            model_dir (str): Directory where downloaded models are stored
+            model_name (str): name of the model to use. Must match the directory name it is stored in, e.g., 'birdnet_default'.
+            preprocessor_config (dict, optional): Keyword arguments for the preprocessor of the model. Defaults to {}.
+            model_config (dict, optional): Keyword arguments for the model instance. Defaults to {}.
+            recording_config (dict, optional): Keyword arguments for the internal SparrowRecording. Defaults to {}.
+            species_predictor_config (dict, optional): Keyword arguments for a species presence predictor model. Defaults to {}.
+            check_time(int, optional): Sleep time of the thread between checks for new files in seconds. Defaults to 1.
+            delete_recordings(str, optional): Mode for data clean up. Can be one of "never", "on_cleanup", "immediatelly". "never" keeps recordings around indefinitely, "on_cleanup" only deletes them when the `clean_up` method is called, and 'immediatelly' deletes the recording immediatelly after analysis.
+        Raises:
+            ValueError: When the indir parameter is not an existing directory.
+            ValueError: When the outdir parameter is not an existing directory.
+            ValueError: When the model_dir parameter is not an existing directory.
+            ValueError: When the model name does not correspond to a directory in the 'model_dir' directory in which available models are stored.
+        """
+        # set up data to use
+        self.input = Path(indir)
+
+        if self.input.is_dir() is False:
+            raise ValueError("Input directory does not exist")
+
+        self.outdir = outdir
+
+        if self.outdir.is_dir() is False:
+            raise ValueError("Output directory does not exist")
+
+        # this is created on disk in the start method
+        self.output = Path(self.outdir) / Path(datetime.now().strftime("%y%m%d_%H%M%S"))
+
+        self.model_dir = Path(model_dir)
+
+        if self.model_dir.is_dir() is False:
+            raise ValueError("Model directory does not exist")
+
+        if (self.model_dir / model_name).is_dir() is False:
+            raise ValueError("Given model name does not exist in model directory")
+
+        self.pattern = pattern
+
+        self.model_name = model_name
+
+        self.watcher_process = None
+
+        self.may_do_work = multiprocessing.Event()
```

Comment:
these 2 events are here to indicate when the analyzer is done with the current file (`is_done_analyzign`, needed to not create corrupt analysis files) and for stopping it from picking up a new one when it should be restarted, or to allow it to be paused (`may_do_work`).